index.js

```diff
@@ -17,20 +42,82 @@ module.exports.file = options => {
 			throw new Error('The `name` and `extension` options are mutually exclusive');
 		}
 
-		return path.join(module.exports.directory(), options.name);
+		return path.join(directory, options.name);
+	}
+
+	return path.join(directory, uniqueString() + '.' + options.extension.replace(/^\./, ''));
+};
+
+module.exports.file = options => getFile(getPath(), options);
+
+module.exports.fileAsync = async options => getFile(await getPathAsync(), options);
+
+module.exports.directory = getPath;
+
+module.exports.directoryAsync = getPathAsync;
+
+module.exports.clean = () => {
```

Comment:
Should `.clean()` really be able to throw? Should users have to think about that?